File: tests/problems/batch/cms-grader/graders/grader.py
import sys

sys.modules["grader"] = sys.modules[__name__]

called_graderfunc = False


def graderfunc():
    global called_graderfunc
    called_graderfunc = True


assert input() == "input"

import gradertest

gradertest.userfunc()

if called_graderfunc:
    print("called graderfunc")

File: tests/problems/batch/cms-grader/graders/grader.cpp
#include "gradertest.h"
#include <cstdlib>
#include <iostream>
#include <string>

namespace
{
    bool called_graderfunc = false;
}

void graderfunc()
{
    ::called_graderfunc = true;
}

int main()
{
    std::string str;
    std::cin >> str;

    if (str != "input")
        std::abort();

    userfunc();

    if (called_graderfunc)
        std::cout << "called graderfunc\n";
}
